src/lightning_app/core/work.py

```diff
@@ -316,9 +316,16 @@ def num_successes(self) -> int:
 
         return has_succeeded_counter
 
+    def _get_property_if_exists(self, name: str) -> Union[property, None]:
+        return getattr(self.__class__, name, None)
+
     def __setattr__(self, name: str, value: Any) -> None:
-        setattr_fn = getattr(self, "_setattr_replacement", None) or self._default_setattr
-        setattr_fn(name, value)
+        property_object = self._get_property_if_exists(name)
```

Comment:
```suggestion
        property_object = getattr(self.__class__, name, None)
```

Reply:
Resolved in https://github.com/Lightning-AI/lightning/pull/14259/commits/30892c892313d386c4c44d368d4bfa20d9d997c0